src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -449,4 +461,40 @@ private void checkCursorAny(Cursor c, CursorKind... kinds) {
         }
         throw new IllegalArgumentException("Invalid cursor kind");
     }
+
+    private <D extends Declaration> D withDeclarationString(D decl, Cursor cursor) {
+        String declString = switch (decl) {
+            case Declaration.Constant _ -> null; // do nothing for enum constants
+            case Typedef _ -> declarationString(cursor, true); // always expand typedefs
```

Comment:
typedefs are always printed in the expanded form because (a) it's not too annoying anyway and (b) in the more compact form, if the entity being typedeffed is an anonymous struct/union/enum, the output of the pretty printer can be weird. For instance in this case:

```
typedef struct { int x; int y; } Y;
```

The pretty printer generates:

```
typedef struct Y Y
```

Which is not great. The expanded form instead prints:

```
typedef struct {
    int x;
    int y;
} Y
```

as expected.